pkg/virt-handler/vm.go

```diff
@@ -691,7 +691,25 @@ type NetworkStatus struct {
 	Interface string   `yaml:"interface"`
 }
 
-func (d *VirtualMachineController) migrationTargetUpdateVMIStatus(vmi *v1.VirtualMachineInstance, domainExists bool) error {
+func domainIsActiveOnTarget(domain *api.Domain) bool {
+
+	if domain == nil {
+		return false
+	}
+
+	// It's possible for the domain to be active on the target node if the domain is
+	// 1. Running
+	// 2. User initiated Paused
+	if domain.Status.Status == api.Running {
+		return true
+	} else if domain.Status.Status == api.Paused && domain.Status.Reason == api.ReasonPausedUser {
+		return true
+	}
+	return false
+
+}
+
+func (d *VirtualMachineController) migrationTargetUpdateVMIStatus(vmi *v1.VirtualMachineInstance, domainExists bool, domain *api.Domain) error {
```

Comment:
@vladikr it would be good to get your thoughts on this change. The purpose of this is to determine when the domain is active on the target during a live migration. For example, when post copy is used, it's possible for the domain to be active before the migration completes.

Do you think the logic here cover us? I'm checking if the domain is running (for any reason) and if the domain is paused due to the user (live migrating a paused vm). Does that make sense, or should I make the paused logic a little broader, and consider the domain active if the paused status is detected on the domain for any reason _other_ than migration `if status == paused && reason != migration; then true`

Reply:
@davidvossel this makes sense to me. The only thing is that I would separate the `isolation.AdjustQemuProcessMemoryLimits` away from `d.finalizeMigration(vmi)`

Previously we've called `finalizeMigration` when we just detected the domain on the target virt-handler,  so `AdjustQemuProcessMemoryLimits` was running before the domain was actually running.

I'm not 100% sure what code path excersises this function though... I think it was needed for sriov hot plug. Perhaps we don't have a good test coverage for this.

Reply:
makes sense. I updated the pr and split the logic up a bit